src/workloads/query/AggregationsOutput.yml

```diff
@@ -0,0 +1,520 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This test exercises both $out and $merge aggregation stages. It does this based on all combinations of the following:
+    A. The src collection is sharded or unsharded
+    B. The document size is either 1K, 4K, or 16K.
+    C. There are 1 thread, 4 threads, or 16 threads doing the same operation.
+  Each actor does the following operations concurrently with other actors at the same stage:
+    A. $out to a collection unique amongst actors.
+       This measures $out performance.
+    B. $merge to the source collection with an updated field value.
+       This measures $merge contention on the same collection.
+    C. $merge to a different unsharded collection unique for each Actor with an updated field value.
+       This measures $merge contention on a separate collection shared by all threads of an Actor.
+
+GlobalDefaults:
+  Unsharded1K: &Unsharded1K Unsharded1K
+  Unsharded4K: &Unsharded4K Unsharded4K
+  Unsharded16K: &Unsharded16K Unsharded16K
+  Sharded1K: &Sharded1K Sharded1K
+  Sharded4K: &Sharded4K Sharded4K
+  Sharded16K: &Sharded16K Sharded16K
+  Sharded1KNSS: &Sharded1KNSS test.Sharded1K
+  Sharded4KNSS: &Sharded4KNSS test.Sharded4K
+  Sharded16KNSS: &Sharded16KNSS test.Sharded16K
+  NumDocs: &NumDocs 10_000
+  Database: &Database test
+  NumPhases: &NumPhases 80
+  NumCombinations: &NumCombinations 18 # 3 threads * 3 data sizes * 2 collection source types
```

Comment:
Given the nature of this workload, it looks like code generation can be useful.
Maybe we can just ping performance tooling team to consider creating an API for code-generated workloads, like `./run-genny generate` or something.

Reply:
+1 on this. Let's ask the perf team if there's a better way to express this.

Reply:
That's a reasonable thing to ask. Most of the pains here revolved around computing offsets for all the combinations in the actor template below.

Reply:
I've opened DEVPROD-3567 for this since there are currently at least two different ways to generate workloads that are completely independent from each other.